README.md

```diff
@@ -1 +1,168 @@
+* `capacity` (uint, optional, default=100) the capacity of the resource
+
+In the default deployment, this configuration shall be provided through a
+config map, for [example](examples/macvtap-deviceplugin-config.yaml):
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: macvtap-deviceplugin-config
+data:
+  DP_MACVTAP_CONF: |
+    [ {
+        "name" : "dataplane",
+        "master" : "eth0",
+        "mode": "bridge",
+        "capacity" : 50
+    } ]
+```
+
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/kubevirt/macvtap-cni/master/examples/macvtap-deviceplugin-config.yaml
+configmap "macvtap-deviceplugin-config" created
+```
+
+This configuration will result in up to 50 macvtap interfaces being offered for
+consumption, using eth0 as the lower device, in bridge mode, and under
+resource name `dataplane.macvtap.network.kubevirt.io`.
+
+The macvtap CNI can be deployed using the proposed
+[daemon set](manifests/macvtap.yaml):
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubevirt/macvtap-cni/master/manifests/macvtap.yaml
+daemonset "macvtap-cni" created
+
+$ kubectl get pods
+NAME                                 READY     STATUS    RESTARTS   AGE
+macvtap-cni-745x4                      1/1    Running           0    5m
+```
+
+This will result in the CNI being installed and device plugin running on all
+nodes.
+
+There is also a [template](templates/macvtap.yaml.in) available to parameterize
+the deployment with different configuration options.
+
+## Usage
+
+macvtap CNI is best used by defining a NetworkAttachmentDefinition:
+
+```yaml
+kind: NetworkAttachmentDefinition
+apiVersion: k8s.cni.cncf.io/v1
+metadata:
+  name: dataplane
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: dataplane.macvtap.network.kubevirt.io
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "name": "dataplane",
+      "type": "macvtap-cni"
+      "mtu": 1500
+    }'
+```
+
+The CNI config json allows the following parameters:
+* `name`     (string, required): the name of the network. When used within a
+   NetworkAttachmentDefinition it can be ommited and will adopt its name. 
+* `type`     (string, required): "macvtap".
+* `mac`      (string, optional): mac address to assign to the macvtap interface.
+* `mtu`      (integer, optional): mtu to set in the macvtap interface.
+* `deviceID` (string, optional): name of an existing macvtap host interface, which
+  will be moved to the correct net namespace and configured. This is typically provided
+  by multus or a similar plugin.
+
+A pod can be attached to that network which would result in the pod having that
+macvtap interface as a second interface:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    k8s.v1.cni.cncf.io/networks: dataplane
+spec:
+  containers:
+  - name: busybox
+    image: busybox
+    command: ["/bin/sleep", "1800"]
+    resources:
+      limits:
+        dataplane.macvtap.network.kubevirt.io: 1 
```

Comment:
Is this really the name under which we expose a resource? AFAIK, usually a DP would use `macvtap.network.kubevirt.io/dataplane` format. Just to keep name of the resource and its kind/namespace.

https://github.com/kubevirt/macvtap-cni/blob/master/tests/e2e/macvtap_test.go#L233

Reply:
oops, you are right